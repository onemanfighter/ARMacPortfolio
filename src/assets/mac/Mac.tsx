import React from 'react';
import { MAC_SVG } from './constants';

const MacIcon = () => {
  return (
    <div
      style={{
        color: '#fff',
      }}
      dangerouslySetInnerHTML={{ __html: MAC_SVG }}
    />
  );
};

export default React.memo(MacIcon);
