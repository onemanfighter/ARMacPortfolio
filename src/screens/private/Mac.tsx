import { Outlet } from 'react-router-dom';
import { Button } from '@chakra-ui/react';
import { modalSelector, useShallow } from '@uiStore/selector';
import { ModalID, uiStore } from '@uiStore';

const Mac = () => {
  const { openModal } = uiStore(useShallow(modalSelector));

  const openModalHandler = () => {
    openModal(ModalID.SEARCH, () => {});
  };
  return (
    <>
      <div>MacScreen</div>
      <Button onClick={openModalHandler}>Open Modal</Button>
      <Outlet />
    </>
  );
};

export default Mac;
