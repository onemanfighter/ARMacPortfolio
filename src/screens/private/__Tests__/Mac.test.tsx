import { render, screen } from '@testing-library/react';
import Mac from '../Mac';

describe('Mac', () => {
  it.skip('should render mac ', () => {
    const { container } = render(<Mac />);

    expect(container).toMatchSnapshot();
  });
});
