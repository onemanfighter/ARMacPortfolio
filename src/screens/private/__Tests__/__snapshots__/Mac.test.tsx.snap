// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mac should render mac  1`] = `
<div>
  <div
    style="filter: brightness(1);"
  >
    <div
      aria-label="mac"
      style="transition: all 0.4s ease-in-out;"
    />
  </div>
</div>
`;
